sql-to-dbsp-compiler/lib/sqllib/src/string.rs

```diff
@@ -318,3 +320,217 @@ pub fn writelog<T: std::fmt::Display>(format: String, argument: T) -> T {
+    }
+
+    #[test]
+    fn test_chr_(){
+        assert_eq!(chr_(65), "A");
+    }
+
+    #[test]
+    fn test_repeat__(){
+        assert_eq!(repeat__("Postgre".to_string(), 4), "PostgrePostgrePostgrePostgre".to_string());
+        assert_eq!(repeat__("Postgre".to_string(), 0), "".to_string());
+    }
+
+    #[test]
+    fn test_overlay3___(){
+        assert_eq!(overlay3___("Txxxxas".to_string(), "hom".to_string(), 2), "Thomxas".to_string());
+    }
+
+    #[test]
+    fn test_overlay4___(){
+        assert_eq!(overlay4____("Txxxxas".to_string(), "hom".to_string(), 2, 4), "Thomas".to_string());
+    }
+
+    #[test]
+    fn test_lower_(){
+        assert_eq!(lower_("Thomas".to_string()), "thomas".to_string());
+    }
+
+    #[test]
+    fn test_upper_(){
+        assert_eq!(upper_("Thomas".to_string()), "THOMAS".to_string());
+    }
+
+    #[test]
+    fn test_initcap_(){
+        assert_eq!(initcap_("hi THOMAS".to_string()), "Hi Thomas".to_string());
+    }
+
+    #[test]
+    fn test_replace___(){
+        assert_eq!(replace___("hi Thomas".to_string(), "hi".to_string(), "Hello".to_string()), "Hello Thomas".to_string());
+    }
+
+    #[test]
+    fn test_left__(){
+        assert_eq!(left__("hi Thomas".to_string(), 2), "hi".to_string());
+    }
+
+    #[test]
+    fn test_split2__(){
+        assert_eq!(split2__("hi Thomas".to_string(), " ".to_string()), vec!["hi".to_string(), "Thomas".to_string()]);
+    }
+
+    #[test]
+    fn test_split1_(){
+        assert_eq!(split1_("hi,Thomas".to_string()), vec!["hi".to_string(), "Thomas".to_string()]);
+    }
+
+    #[test]
+    fn test_array_to_string2_vec__(){
+        assert_eq!(array_to_string2_vec__(vec!["John".to_string(), "Thomas".to_string()], ",".to_string()), "John,Thomas".to_string());
+    }
+
+    #[test]
+    fn test_array_to_string2Nvec__(){
+        assert_eq!(array_to_string2Nvec__(vec![Some("John".to_string()), None, Some("Thomas".to_string())], " ".to_string()), "John Thomas".to_string());
+    }
+
+    #[test]
+    fn test_array_to_string3_vec___(){
+        assert_eq!(array_to_string3_vec___(vec!["John".to_string(), "Thomas".to_string()], " ".to_string(),"".to_string()), "John Thomas".to_string());
+    }
+
+    #[test]
+    fn test_array_to_string3Nvec___(){
+        assert_eq!(array_to_string3Nvec___(vec![Some("John".to_string()), None, Some("Thomas".to_string())], " ".to_string(),"None".to_string()), "John None Thomas".to_string());
+    }
+
+    #[test]
+    fn test_to_lower_hex(){
+        assert_eq!(to_lower_hex(2147483647), "7fffffff");
+    }
+
+    #[test]
+    fn test_to_upper_hex(){
+        assert_eq!(to_upper_hex(2147483647), "7FFFFFFF");
+    }
+
+    #[test]
+    fn test_md5__(){
+        assert_eq!(md5__("Value".to_string()), "689202409e48743b914713f96d93947c".to_string());
+    }
+
+    #[test]
+    fn test_translate_(){
+        assert_eq!(translate_("12345".to_string(), "143".to_string(), "ax".to_string()), "a2x5".to_string());
```

Comment:
I guess I see what it does now here, probably hard to find a library function for this, but good to check anyways,